app/src/main/java/com/github/k409/fitflow/ui/screens/profile/ProfileViewModel.kt

```diff
@@ -10,18 +10,21 @@ class ProfileViewModel @Inject constructor(
     private val profileRepository: ProfileRepository,
 ) : ViewModel() {
     fun submitProfile(
+        uid: String,
         name: String,
         age: Int,
         gender: String,
         weight: Int,
         height: Int,
-    ) {
+    ): Boolean {
         profileRepository.submitProfile(
+            uid,
             name,
             age,
             gender,
             weight,
             height,
         )
+        return profileRepository.success
```

Comment:
If you make the `submitProfile` method in the repository `suspend`, make this method suspend also.